### PR TITLE
Add UPS client (netclient) support in NUT plugin

### PR DESCRIPTION
**Pull request by daniel0076**
Fixes: #318

The patch is for OMV5.
Tested with my UPS on local standalone mode, local standalone with remote monitor.
And UPS on DS918+ for netclient mode.

Signed-off-by: Daniel Tsai <idaniel.twc@gmail.com
- [x] References issue
- [ ] Includes tests for new functionality or reproducer for bug

### CODE REVIEW

File: deb/openmediavault-nut/usr/share/openmediavault/datamodels/conf.service.nut.json [hunk 1]
```diff
@@ -11,13 +11,34 @@
...
+			"default": "standalone"
+		},
 		"upsname": {
 			"type": "string",
 			"pattern": "^[a-z0-9_-]+$"
 		},
 		"comment": {
 			"type": "string"
 		},
+		"nutserver": {
```
Comment (votdev) [inline deb/openmediavault-nut/usr/share/openmediavault/datamodels/conf.service.nut.json:26]:
```suggestion
		"remotehostname": {
```

File: deb/openmediavault-nut/usr/share/openmediavault/datamodels/conf.service.nut.json [hunk 2]
```diff
@@ -11,13 +11,34 @@
...
+			"type": "string"
+		},
+		"nutserverlogin": {
+			"type": "string",
+			"default": ""
+		},
+		"nutserverpassword": {
```
Comment (votdev) [inline deb/openmediavault-nut/usr/share/openmediavault/datamodels/conf.service.nut.json:33]:
```suggestion
		"remotepassword": {
```

File: deb/openmediavault-nut/var/www/openmediavault/js/omv/module/admin/service/nut/Nut.js [hunk 3]
```diff
@@ -85,13 +140,40 @@ Ext.define("OMV.module.admin.service.nut.Settings", {
...
 				allowBlank: true
+			},{
+				xtype: "textfield",
+				name: "nutserver",
+				fieldLabel: _("NUT server address"),
+				allowBlank: false
+			},{
+				xtype: "textfield",
+				name: "nutserverlogin",
+				fieldLabel: _("NUT server login"),
```
Comment (votdev) [inline deb/openmediavault-nut/var/www/openmediavault/js/omv/module/admin/service/nut/Nut.js:158]:
```suggestion
				fieldLabel: _("Remote server username"),
```

File: deb/openmediavault-nut/usr/share/openmediavault/datamodels/conf.service.nut.json (hunk 2 above)
Comment (daniel0076) [reply deb/openmediavault-nut/usr/share/openmediavault/datamodels/conf.service.nut.json:33]: `remotepassword` is already taken and used for remote monitor under standalone mode.
Perhaps `netclientpassword` will be better?

https://github.com/openmediavault/openmediavault/blob/811229a7c9e90682df1c17849f1ddcbe0948947a/deb/openmediavault-nut/usr/share/openmediavault/datamodels/conf.service.nut.json#L59-L64

File: deb/openmediavault-nut/usr/share/openmediavault/datamodels/conf.service.nut.json [hunk 4]
```diff
@@ -11,13 +11,34 @@
...
 			"type": "string",
 			"pattern": "^[a-z0-9_-]+$"
 		},
 		"comment": {
 			"type": "string"
 		},
+		"nutserver": {
+			"type": "string"
+		},
+		"nutserverlogin": {
```
[the remark this answers is not in the document]
Comment (daniel0076) [reply deb/openmediavault-nut/usr/share/openmediavault/datamodels/conf.service.nut.json:29]: Same as above
`remoteuser` is already taken and used for remote monitor under standalone mode.
Using `remoteusername` may cause confusion.
Perhaps `netclientlogin` or `netclientuser` ?

https://github.com/openmediavault/openmediavault/blob/811229a7c9e90682df1c17849f1ddcbe0948947a/deb/openmediavault-nut/usr/share/openmediavault/datamodels/conf.service.nut.json#L59-L64

[the remark this answers is not in the document]
Comment (votdev) [reply deb/openmediavault-nut/usr/share/openmediavault/datamodels/conf.service.nut.json:29]: Please use `netclientxxxx` everywhere in that case.

[the remark this answers is not in the document]
Comment (daniel0076) [reply deb/openmediavault-nut/usr/share/openmediavault/datamodels/conf.service.nut.json:29]: Okay, will be `netclienthostname`, `netclientusername`, `netclientpassword` respectively.
I agree with other requests to change, working on them :)

File: deb/openmediavault-nut/usr/share/openmediavault/datamodels/rpc.nut.json [hunk 5]
```diff
@@ -17,6 +22,22 @@
...
+			"nutserverpassword": {
+				"type": "string",
+				"required": true
+			},
+			"powervalue": {
+				"type": "integer",
+				"required": true
+			},
 			"driverconf": {
 			    "type": "string",
```
[the remark this answers is not in the document]
Comment (daniel0076) [reply deb/openmediavault-nut/usr/share/openmediavault/datamodels/rpc.nut.json:42]: I found some other indention with spaces in this file from previous commits.
Fix them to tabs as well.

**Comment by daniel0076** [comment]
The request to changes are resolved with a new  force push commit.

**Comment by votdev** [comment]
A config migration script like this https://github.com/openmediavault/openmediavault/blob/master/deb/openmediavault-shairport/usr/share/openmediavault/confdb/migrations.d/conf.service.shairport_2.5.sh is missing. The new fields must be added, otherwise existing installations will fail.

**Comment by daniel0076** [comment]
The config migration script is added.
I am not sure what the next version of the plugin will be, so currently I arbitrarily name the script `conf.service.nut_5.4.1.sh`. Please kindly correct me.

https://github.com/openmediavault/openmediavault/commit/1188fc7fbb21c63d0e6408ec39faff3dab79d048#diff-b757b71ccaf4cac901d04b0c3f27901b

**Comment by votdev** [comment]
I will close this in favor of https://github.com/openmediavault/openmediavault/pull/681. Have to do some fixes to finalize the PR. Thank you for your contribution.

**Comment by daniel0076** [comment]
Okay, thanks for that.

**Pull request closed without merging**